File: Solutions/152. Maximum Product Subarray/152. Maximum Product Subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int res=nums[0],l=0,r=0,n=nums.length;
        for(int i=0;i<n;++i){
            l=(l!=0?l:1)*nums[i];
            r=(r!=0?r:1)*nums[n-i-1];
            res=Math.max(res,Math.max(l,r));
        }
        return res;
    }
}

File: Solutions/152. Maximum Product Subarray/152. Maximum Product Subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int n=nums.size(),l=0,r=0,res=INT_MIN;
        for(int i=0;i<n;++i){
            l=(l==0?1:l)*nums[i];
            r=(r==0?1:r)*nums[n-i-1];
            res=max({res,l,r});
        }
        return res;
    }
};